### PR TITLE
Add article hover buttons - View, Ophan, Delete

### PR DESCRIPTION
**Pull request by vlbee**
WIP for code review only. 

- Created a `ButtonHoverAction` component in `shared/components/inputs` that extends `ButtonCircular`
- This button component takes in a prop to determine what hover "action" it will perform - eg. link to Ophan, delete, copy article etc. 
- Component is being used within `shared/components/Article` inside the `HoverActions` component (per conversation with @RichieAHB this itself could be turned into a reusable component.)

- Hover Icons svgs imported (not final design files).  Will be unique designs, hence why different svgs to other Icon svgs.)

TODO: 

- Add external links for Ophan and article preview link buttons. 
I am unsure where to find the url data for the links - is this already is being passed into the Fronts article as a prop?

QUESTION:
Colors - is there some sort of global styles context or variables that could be used?

### CODE REVIEW

File: client-v2/src/shared/images/icons-hover/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import View from './icon-hover-view.svg';
+import Ophan from './icon-hover-ophan.svg';
+import Copy from './icon-hover-copy.svg';
+import Paste from './icon-hover-paste-sublink.svg';
+import Clipboard from './icon-hover-clipboard.svg';
+import Delete from './icon-hover-delete.svg';
+
+const hoverActionIcons = {
```
Comment (RichieAHB) [inline client-v2/src/shared/images/icons-hover/index.js]: We could export these as named exports like `export { default as view } from './icon-hover-view.svg';` which just means that webpack will tree shake any unused SVGs in future. When they're exported on an object then webpack can't do this. It'll probably be fine but perhaps worth guarding against.

File: client-v2/src/shared/components/input/ButtonHoverAction.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// @flow
+import * as React from 'react';
+import styled from 'styled-components';
+import ButtonCircular from './ButtonCircular';
+import hoverActionIcons from '../../images/icons-hover/index';
+
+const Link = styled(`a`).attrs({
+  target: '_blank',
+  rel: 'noopener noreferrer'
```
[the remark this answers is not in the document]
Comment (jonathonherbert) [reply client-v2/src/shared/components/input/ButtonHoverAction.js]: Agreed, awesome! 👍

File: client-v2/src/shared/images/icons-hover/index.js (hunk 1 above)
Comment (vlbee) [reply client-v2/src/shared/images/icons-hover/index.js]: I have made this change.

File: client-v2/src/shared/components/Article.js [hunk 3]
```diff
@@ -249,16 +240,40 @@ const ArticleComponent = ({
...
-          >
-            ✕
-          </ActionButton>
+          <HoverActionsLeft>
+            <ButtonHoverAction
+              action="view"
+              onClick={e => {
+                // stop the parent from opening the edit panel
+                e.stopPropagation();
+                // TODO click action
```
[the remark this answers is not in the document]
Comment (vlbee) [reply client-v2/src/shared/components/Article.js]: Not 100% how to do this but can look at refactoring when adding the rest of the copy/paste buttons.

**Comment by vlbee** [comment]
Completed adding View, Ophan and Delete hover buttons - https://trello.com/b/lN3dWiWJ/fronts

Flow coverage is a bit low.

**Comment by vlbee** [comment]
I would like to try to improve type coverage on Article.js before merging.

**Pull request merged**